tests/e2e/provisioning/pkg/client/broker/broker_client.go

```diff
@@ -0,0 +1,229 @@
+		globalAccountID: globalAccountID,
+		client:          &clientHttp,
+		log:             log,
+	}
+}
+
+const (
+	kymaClassID = "47c9dcbf-ff30-448e-ab36-d3bad66ba281"
+	gcpPlanID   = "ca6e5357-707f-4565-bbbd-b3ab732597c6"
+	azurePlanID = "4deee563-e5ec-4731-b9b1-53b42d855f0c"
+
+	instancesURL = "/v2/service_instances"
+)
+
+func (c *Client) ProvisionRuntime() (string, error) {
+	c.log.Infof("Provisioning Runtime [ID: %s, NAME: %s]", c.runtimeID, c.clusterName)
+	requestByte, err := c.prepareProvisionDetails()
+	if err != nil {
+		return "", errors.Wrap(err, "while marshalling request body")
+	}
+	c.log.Infof("Provisioning parameters: %v", string(requestByte))
+
+	provisionURL := fmt.Sprintf("%s%s/%s", c.brokerConfig.URL, instancesURL, c.runtimeID)
+	response := provisionResponse{}
+	err = wait.Poll(time.Second, time.Second*5, func() (bool, error) {
+		err := c.executeRequest(http.MethodPut, provisionURL, bytes.NewReader(requestByte), &response)
+		if err != nil {
+			c.log.Warn(errors.Wrap(err, "while executing request").Error())
+			return false, nil
+		}
+		if response.Operation == "" {
+			c.log.Warn("Got empty operation ID")
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return "", errors.Wrap(err, "while waiting for successful provision call")
+	}
+	c.log.Infof("Successfully send provision request, got operation ID %s", response.Operation)
+
+	return response.Operation, nil
+}
+
+func (c *Client) DeprovisionRuntime() error {
+	deprovisionURL := fmt.Sprintf("%s%s/%s", c.brokerConfig.URL, instancesURL, c.runtimeID)
+
+	response := provisionResponse{}
+	c.log.Infof("Deprovisioning Runtime [ID: %s, NAME: %s]", c.runtimeID, c.clusterName)
+	err := wait.Poll(time.Second, time.Second*5, func() (bool, error) {
+		err := c.executeRequest(http.MethodDelete, deprovisionURL, nil, &response)
+		if err != nil {
+			c.log.Warn(errors.Wrap(err, "while executing request").Error())
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return errors.Wrap(err, "while waiting for successful deprovision call")
+	}
+	c.log.Infof("Successfully send deprovision request, got operation ID %s", response.Operation)
+	return nil
+}
+
+func (c *Client) AwaitProvisioningSucceeded(operationID string) error {
+	lastOperationURL := fmt.Sprintf("%s%s/%s/last_operation?operation=%s", c.brokerConfig.URL, instancesURL, c.runtimeID, operationID)
+
+	response := lastOperationResponse{}
+	err := wait.Poll(time.Minute, c.brokerConfig.ProvisionTimeout, func() (bool, error) {
+		err := c.executeRequest(http.MethodGet, lastOperationURL, nil, &response)
+		if err != nil {
+			c.log.Warn(errors.Wrap(err, "while executing request").Error())
+			return false, nil
+		}
+		switch domain.LastOperationState(response.State) {
+		case domain.Failed:
+			c.log.Info("Provisioning failed")
+			return true, errors.New("provisioning failed")
+		case domain.Succeeded:
+			c.log.Infof("Runtime Provisioning succeeded!")
+			return true, nil
+		default:
```

Comment:
handle also `domain.InProgress` state, if in some reason there will be an empty response ("{}") then the state will be an empty string, we can add log about it just in case (in the default case)